process.go

```diff
@@ -391,14 +391,18 @@ func fastDraw(dst *image.NRGBA, src *image.NRGBA, x, y int) {
 	maxY := bounds.Max.Y
 	maxX := bounds.Max.X * 4
 
+	srcColor := [4]float32{}
+	dstColor := [4]float32{}
+	finalColor := [4]uint8{}
+
 	pointer := dst.PixOffset(x, y)
 	for row := 0; row < maxY; row += 1 {
 		for i := 0; i < maxX; i += 4 {
 			srcPx := row*src.Stride + i
 			dstPx := row*dst.Stride + i + pointer
 
-			srcColor := colorToFloat(src, srcPx)
-			dstColor := colorToFloat(dst, dstPx)
+			uintToFloat(src, srcPx, &srcColor)
```

Comment:
Reason I use floats is becuause its common in glsl to use it for colors and also makes more sense in this context for maths

We dont really have to use floats though we can do something like this
Casting to int then back  which is less readable I think
```go
sR := int(src.Pix[srcPx])
sA := int(src.Pix[srcPx+3])

dR := int(dst.Pix[dstPx])

dst.Pix[dstPx+0] = uint8(((sR * sA) / 255) + (dR * (255 - sA) / 255))
```